subiquity/models/subiquity.py

```diff
@@ -314,6 +322,20 @@ def _cloud_init_files(self):
                 ('etc/hostname', hostname + "\n", 0o644),
                 ('etc/hosts', HOSTS_CONTENT.format(hostname=hostname), 0o644),
                 ])
+        # Add cloud-init clean hooks to support golden-image creation.
+        cfg_files = ["/" + path for (path, _content, _cmode) in files]
+        network_write_files = self.network.render().get('write_files').values()
```

Comment:
I think I'd rather run this in reverse, that `network` is responsible for registering that there is a file to remove.  A quick new function on this model to append to a list of files to be cleaned ought to do the trick.

Reply:
Sounds good to me. hopefully I understood  this suggestion properly  network model should announce/register it's emitted config files which can be removed. For lack of a better name I called the method `NetworkModel.rendered_config_paths`

Reply:
Yes, this works.  If we have other similar cases we can add a `rendered_config_paths` to other models.  Thanks!